baselines/flwr_baselines/scripts/leaf/reddit/split_json_data.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2021 Adap GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import argparse
+import json
+import pickle
+from os import PathLike
+from pathlib import Path
+from typing import List, Optional, Tuple
+
+
+def check_between_zero_and_one(value: str):
+    fvalue = float(value)
+    if fvalue < 0 or fvalue > 1:
+        raise argparse.ArgumentTypeError(
+            f"""Invalid partition fraction {fvalue}. This must be between 0 and 1."""
+        )
+    return fvalue
+
+
+def split_json_and_save(
+    list_datasets: List[Tuple[str, float]],
+    path_to_json: PathLike,
+    save_root: PathLike,
+    users_list: Optional[List[str]] = [],
```

Comment:
Using mutable default arguments in Python is usually [discouraged](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments).

Better:

```python
users_list: Optional[List[str]] = Null
```
and in the function body
```python
if users_list is Null
  users_list = []
```